sycl/include/CL/sycl/aliases.hpp

```diff
@@ -0,0 +1,115 @@
+//==----------- aliases.hpp --- SYCL type aliases --------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/detail/common.hpp>
+
+#include <cstddef>
+#include <cstdint>
+
+namespace cl {
+namespace sycl {
+template <typename T, int N> class vec;
+namespace detail {
+namespace half_impl {
+class half;
+} // namespace half_impl
+} // namespace detail
+} // namespace sycl
+} // namespace cl
+
+#ifdef __SYCL_DEVICE_ONLY__
+using half = _Float16;
+#else
+using half = cl::sycl::detail::half_impl::half;
+#endif
+
+#define MAKE_VECTOR_ALIAS(ALIAS, TYPE, N)                                      \
+  using ALIAS##N = cl::sycl::vec<TYPE, N>;
+
+#define MAKE_VECTOR_ALIASES_FOR_ARITHMETIC_TYPES(N)                            \
+  MAKE_VECTOR_ALIAS(char, char, N)                                             \
+  MAKE_VECTOR_ALIAS(short, short, N)                                           \
+  MAKE_VECTOR_ALIAS(int, int, N)                                               \
+  MAKE_VECTOR_ALIAS(long, long, N)                                             \
+  MAKE_VECTOR_ALIAS(float, float, N)                                           \
+  MAKE_VECTOR_ALIAS(double, double, N)                                         \
+  MAKE_VECTOR_ALIAS(half, ::half, N)
+
+#define MAKE_VECTOR_ALIASES_FOR_OPENCL_TYPES(N)                                \
+  MAKE_VECTOR_ALIAS(cl_char, cl::sycl::cl_char, N)                             \
+  MAKE_VECTOR_ALIAS(cl_uchar, cl::sycl::cl_uchar, N)                           \
+  MAKE_VECTOR_ALIAS(cl_short, cl::sycl::cl_short, N)                           \
+  MAKE_VECTOR_ALIAS(cl_ushort, cl::sycl::cl_ushort, N)                         \
+  MAKE_VECTOR_ALIAS(cl_int, cl::sycl::cl_int, N)                               \
+  MAKE_VECTOR_ALIAS(cl_uint, cl::sycl::cl_uint, N)                             \
+  MAKE_VECTOR_ALIAS(cl_long, cl::sycl::cl_long, N)                             \
+  MAKE_VECTOR_ALIAS(cl_ulong, cl::sycl::cl_ulong, N)                           \
+  MAKE_VECTOR_ALIAS(cl_float, cl::sycl::cl_float, N)                           \
+  MAKE_VECTOR_ALIAS(cl_double, cl::sycl::cl_double, N)                         \
+  MAKE_VECTOR_ALIAS(cl_half, cl::sycl::half, N)
```

Comment:
What cl_half do you mean? OpenCL version or SYCL version?
In SYCL cl_half is 

> Alias to a 16-bit floating-point. The half data type must conform to the IEEE 754-
> 2008 half precision storage format.

So it not the same that is OpenCL cl_half.

```
  {
    using T = cl::sycl::vec</*OpenCL version*/cl_half, 2>;
    T v(0.4, 0.5);
    T::element_type x = v.x();
    std::cout << x << std::endl;
  }
```
  x will be zero because `typedef unsigned __int16 cl_half;` in OpenCL run-time headers.

Reply:
@romanovvlad Does the answer satisfy you?

Reply:
I'm asking how cl::sycl::vec methods will work with cl_half:
```
  {
    using T = cl::sycl::vec<half, 3>;
    T v1(0.4, 0.5, 0.2);
     T v2(0.1, 0.2, 0.4);
     v1 = v1 / v2;
  }
```

Reply:
For both cases v1 is equal {4, 2.5, 0.5}.

Reply:
In our implementation `cl::sycl::cl_half3` is the same type as `cl::sycl::half3`.
In SYCL `half` is one of built-in scalar data types and is a 16-bit floating-point and  conform to the IEEE 754-2008 half precision storage format.
`cl_half` is alias to a 16-bit floating-point. 
So in SYCL `cl::sycl::cl_half` and `half` can be the same type. 
Also the size and alignment of  `cl::sycl::cl_half` coincides with the OpenCL variant `cl_half`(and `half`). 
Therefore, we are using cl::sycl::cl_half = half;

Reply:
Thanks for explanation.